src/libutil/util.cc

```diff
@@ -2026,17 +2035,21 @@ void connect(int fd, const std::string & path)
 {
     struct sockaddr_un addr;
     addr.sun_family = AF_UNIX;
+    // This violates the strict aliasing rule, but the implementation of the
+    // bind function contains some hints to the compiler that allow for this
+    // special case.
+    auto * psaddr {reinterpret_cast<struct sockaddr *>(&addr)};
```

Comment:
I don't like that this comment is repeated. Could you factor the cast and comment into a function?
Perhaps even its own type, to automatically set `AF_UNIX` as well.